runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -101,9 +102,8 @@ public void copyFrom(ParserRuleContext ctx) {
 			this.children = new ArrayList<>();
 			// reset parent pointer for any error nodes
 			for (ParseTree child : ctx.children) {
-				if ( child instanceof ErrorNodeImpl ) {
-					this.children.add(child);
-					((ErrorNodeImpl) child).parent = this;
+				if ( child instanceof ErrorNode ) {
+					addChild((ErrorNode)child);
```

Comment:
The error nodes should be *moved* to the new context in `copyFrom` as it's used only for flipping contexts per the comment.